### ISSUE TITLE
Rule object-shorthand misleads the use of shorthand for get keys

### ISSUE THREAD

**Issue by ricardogama** [eslint/eslint#6408]
**What version of ESLint are you using?**

2.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```json
{
  "env": {
    "es6": true,
    "node": true
  },
  "rules": {
    "object-shorthand": "error"
  }
}

```

**What did you do? Please include the actual source code causing the issue.**

```js
// test.js

var get = 'foo';
var bar = { get: get };
```

```sh
eslint test.js
```

**What did you expect to happen?**

Nothing

**What actually happened? Please include the actual, raw output from ESLint.**

```
error  Expected property shorthand  object-shorthand
```

But due to the getter notation the use of object shorthand with a `get` key throws the following error:

```js
// test.js

var get = 'foo';
var bar = { get };
```

```sh
eslint test.js
```

```
test.js:2
var bar = { get };
                ^

SyntaxError: Unexpected token }
```

**Comment by mysticatea** [comment]
Thank you for this issue.

`object-shorthand` rule can be used in only ES2015 (ES6) and later.
Please try to use with ES6 configuration.

![image](https://cloud.githubusercontent.com/assets/1937871/16041350/caec4664-326f-11e6-8646-8e3ebf508252.png)

**Comment by ricardogama** [comment]
@mysticatea Indeed, but the issue persists. Updated the configuration section on the issue description.

**Comment by mysticatea** [comment]
Hm, but I cannot reproduce this on our online demo and Chrome devtools' console.

**Comment by mysticatea** [comment]
By the language spec, we cannot use reserved words in shorthand properties. But `get` and `set` are not reserved words, so we should be able to use those.

https://tc39.github.io/ecma262/2016/#prod-PropertyDefinition

**Comment by ricardogama** [comment]
@mysticatea I can't reproduce the error on Chrome either, but this definitely happens with Node.js 4.0 and above. What about adding this exception only when **env > node** is enabled?

**Comment by michaelficarra** [comment]
@ricardogama Not true.

```
$ node --version
v6.2.1
$ node
> var get = 5
undefined
> ({get})
{ get: 5 }
>
```

**Comment by ricardogama** [comment]
@michaelficarra 5.x fails as well, only works on 6.x.

**Comment by nzakas** [comment]
Unfortunately, there's nothing we can do here. This is a syntax error in the Node.js version you're using, and we can't help with that (exception comments won't help because we aren't the one throwing the error).

I agree with @michaelficarra, this is a bug in an old version of Node.js, so I don't think there's a good argument for changing the rule.

**Comment by ricardogama** [comment]
I agree, cheers guys!
